### PR TITLE
Explosion and Explosives extension and small fixes

### PR DESCRIPTION
**Pull request by janKSU**
Added hit collision for explosion and explosives
Added killable and life according to the radius to asteroids
Added killable, health and hit function for barrel
Added constant damage for the explosion
Added image preloader for the explosion
Added rounding for the remaining player life

### CODE REVIEW

**Comment by janKSU** [comment]
I will do the cooldown, so please wait with the review until then.

**Review by ksu-shanemat — APPROVED** [review]

**Comment by janKSU** [comment]
I added cooldown for the barrel. It will allow being dropped by player ship without any damage.

File: src/space/objects/realization/barrel.js [hunk 1]
```diff
@@ -2,10 +2,16 @@ import AssetLoader from "../../utils/assetLoader";
...
-import Explosion,{LAST_IMAGE_SUFFIX} from "./explosion";
+import Explosion, {LAST_IMAGE_SUFFIX} from "./explosion";
 
-/** Name of base bomb asset */
-const BARREL_ASSET_NAME = "barrel";
+/** Name of active barrel asset */
+const BARREL_ACTIVE_ASSET_NAME = "barrel";
+
+/** Name of the inactvie barrel asset */
+const BARREL_INACTIVE_ASSET_NAME = "barrel_inactive";
```
Comment (RealJohnSmith) [inline src/space/objects/realization/barrel.js:11]: Corresponding image file is missing.

**Review by RealJohnSmith — CHANGES_REQUESTED** [review]

File: src/space/objects/realization/barrel.js (hunk 1 above)
Comment (janKSU) [reply src/space/objects/realization/barrel.js:11]: Yes, I forgot to upload, sorry!

Comment (janKSU) [reply src/space/objects/realization/barrel.js:11]: It is uploaded

**Review by RealJohnSmith — APPROVED** [review]

**Comment by RealJohnSmith** [comment]
Rebase mergeed

**Pull request closed without merging**